src/KingSystem/Sound/sndMusicInfo.cpp

```diff
@@ -0,0 +1,117 @@
+      mPart2WaitCountRate(1, 3, "Part2WaitCountRate", "倍率", "Min=1,Max=10", &mHeader),
+      mPart2Volume(0.4, 1, "Part2Volume", "音量ランダム幅", "Min=0.f,Max=1.f", &mHeader),
+      // mPart2Pan missing?
+      mPart3WaitCountBase(50, "Part3WaitCountBase", "基本の値", "Min=0,Max=100", &mData),
+      mPart3WaitCountRate(1, 4, "Part3WaitCountRate", "倍率", "Min=1,Max=10", &mHeader),
+      mPart3Volume(0.4, 1, "Part3Volume", "音量ランダム幅", "Min=0.f,Max=1.f", &mHeader),
+      mPart3Pan(-1, 1, "Part3Pan", "パンランダム幅", "Min=-1.f,Max=1.f", &mHeader) {}
+
+void MusicParam::init(sead::Heap* heap) {
+    // mData not linked in any way?
+    addList(&mHeader, "MusicParam");
+    mParamPerTypes.allocBufferAssert(2, heap);
+    mHeader.addObj(&mParamPerTypes[0],
+                   sead::FormatFixedSafeString<32>("ParamPerType_%s", EnvTimeEnum(0).text()));
+    mHeader.addObj(&mParamPerTypes[1],
+                   sead::FormatFixedSafeString<32>("ParamPerType_%s", EnvTimeEnum(1).text()));
+    sead::DirectResource* sound = loadMusic();
+    if (sound) {
+        agl::utl::ResParameterArchive archive(sound->getRawData());
+        applyResParameterArchive(archive);
+    }
+}
+
+HorseBgm::HorseBgm()
+    : mTimeDay(5, "TimeDay", "この時刻以降は昼扱い", "Min=0.f, Max=24.f", &mData),
+      mTimeNight(20, "TimeNight", "この時刻以降は夜扱い", "Min=0.f, Max=24.f", &mData),
+      mPlayWaitSec(5, "PlayWaitSec", "快速ギアになってから再生開始までの待ち秒", "Min=0.f,Max=10.f",
+                   &mData),
+      mFadeInSec(14, "FadeInSec", "フェードイン秒", "Min=0.f,Max=30.f", &mData),
+      mMinVolume(0, "MinVolume", "フェードイン開始時点の音量", "Min=0.f,Max=1.f", &mData),
+      mFadeOutWaitSec(3, "FadeOutWaitSec", "ギアが下がった時のフェードアウト開始待ち秒",
+                      "Min=0.f,Max=10.f", &mData),
+      mFadeOutSec(8, "FadeOutSec", "フェードアウト秒", "Min=0.f,Max=30.f", &mData),
+      mPlayingSecUseOutro(18, "PlayingSecUseOutro",
+                          "フェードアウト完了時に再生がこの秒数以上続いてたらアウトロあり",
+                          "Min=0.f,Max=60.f", &mData),
+      mIgnoreRequestSec(50, "IgnoreRequestSec",
+                        "フェードアウト開始時に発動する開始リクエスト禁止秒", "Min=0.f,Max=100.f",
+                        &mData) {}
+
+void HorseBgm::init(sead::Heap* heap) {
+    addObj(&mData, "MusicParam");
+    sead::DirectResource* sound = loadMusic();
+    if (sound) {
+        agl::utl::ResParameterArchive archive(sound->getRawData());
+        applyResParameterArchive(archive);
+    }
+}
+
+TemperatureBgm::ParamPerType::ParamPerType()
+    : mTemperatureThreshold(30, "TemperatureThreshold", "温度の閾値", "Min=-50.f,Max=50.f", this),
+      mFadeOutSec(2, "FadeOutSec", "条件を外れた時のフェードアウト秒", "Min=0.f,Max=10.f", this),
+      mCancelFadeSec(0.5, "CancelFadeSec", "再度条件に入った時の復帰フェード秒", "Min=0.f,Max=10.f",
+                     this) {}
+
+TemperatureBgm::TemperatureBgm() = default;
+
+void TemperatureBgm::init(sead::Heap* heap) {
+    addList(&mData, "MusicParam");
+    mParamPerTypes.allocBufferAssert(3, heap);
```

Comment:
Do not hardcode the number of params:

```suggestion
    mParamPerTypes.allocBufferAssert(TemperatureEnum::size(), heap);
```